spec/account/account-service.spec.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+describe("AccountService", () => {
+    var accountService;
+
+    beforeEach(() => {
+        var AccountService, config, password, promiseMock;
+
+        AccountService = require("../../lib/account/account-service");
+        promiseMock = require("../mock/promise-mock");
+        class StorageFake {
+            constructor() {
+                this.configure = jasmine.createSpy("storage.configure");
+                this.configure.andCallFake(() => {
+                    return this;
+                });
+                this.getAsync = jasmine.createSpy("getAsync");
+                this.getAsync.andCallFake((params) => {
+                    return promiseMock.resolve(
+                        new Buffer('{"data": "thing"}', "binary")
+                    );
+                });
+                this.putAsync = jasmine.createSpy("putAsync");
+                this.putAsync.andCallFake((params) => {
+                    return promiseMock.resolve(() => {
+                        return true;
+                    });
+                });
+            }
+        }
+        config = {
+            storage: {
+                bucket: "some-place-wonderful"
+            }
+        };
+        password = jasmine.createSpyObj("password", [
+            "hashContent"
+        ]);
+        password.hashContent.andCallFake(() => {
+            return "hashedContent";
```

Comment:
On line 50 I validate the result of passing in the password from what comes back from the service now. From a comment above instead of returning true, we pass back the data of what was sent in.

```
expect(result).toEqual({
    email: "some.one@example.net",
    password: "somereallylonghashedpassword"
});
```